### PR TITLE
Spdm1.2 transcript: Add support to Spdm1.2 transcript (patch 1)

### PR DESCRIPTION
**Pull request by longlongyang**
As per specification of SPDM1.2, change transcript calculation to support
1. SPDMsign(PrivKey, M1, "challenge_auth signing");
2. SPDMsign(PrivKey, L1, "measurements signing");
3. SPDMsignatureVerify(PubKey, Signature, M2, "challenge_auth signing");
4. SPDMsignatureVerify(PubKey, Signature, L2, "measurements signing")

Signed-off-by: Longlong Yang <longlong.yang@intel.com>

### CODE REVIEW

File: spdmlib/src/common/mod.rs [hunk 1]
```diff
@@ -589,6 +674,15 @@ pub struct SpdmNegotiateInfo {
...
     pub key_schedule_sel: SpdmKeyScheduleAlgo,
+    pub opaque_data_support: bool,
+    pub opaque_data_fmt0: bool, // ignore when opaque_data_support is false;
+    // this field shall indicate that the format for
+    // all OpaqueData fields in this specification
+    // is defined by the device vendor or other standards body.
+    pub opaque_data_fmt1: bool, // ignore when opaque_data_support is false;
+                                // this field shall indicate that the format for
+                                // all OpaqueData fields in this specification
+                                // is defined by the General opaque data format.
```
Comment (jyao1) [inline spdmlib/src/common/mod.rs:685]: Please dont use 3 bool.
Please just use 1 field to describe the opaque support.

File: spdmlib/src/common/mod.rs [hunk 2]
```diff
@@ -638,6 +733,15 @@ pub struct SpdmProvisionInfo {
...
     pub default_version: SpdmVersion,
+    pub opaque_data_support: bool,
+    pub opaque_data_fmt0: bool, // Support or not, true for support, if support
+    // this field shall indicate that the format for
+    // all OpaqueData fields in this specification
+    // is defined by the device vendor or other standards body.
+    pub opaque_data_fmt1: bool, // Support or not, true for support, if support
+                                // this field shall indicate that the format for
+                                // all OpaqueData fields in this specification
+                                // is defined by the General opaque data format.
```
Comment (jyao1) [inline spdmlib/src/common/mod.rs:744]: please treat opaque data format as one type of algo.
No need to provision here.

File: spdmlib/src/common/mod.rs (hunk 1 above)
Comment (longlongyang) [reply spdmlib/src/common/mod.rs]: Good point, I created a SpdmOpaqueSupport bitflags variable to do this kind of things. Please check the updated one, Thank you!

File: spdmlib/src/common/mod.rs (hunk 2 above)
Comment (longlongyang) [reply spdmlib/src/common/mod.rs]: > 

Well, please check the updated one, Thank you!

**Pull request merged**